### PR TITLE
Migrate all model dependencies from conda to "pip" section

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes are proposed in this pull request?

Migrate all model dependencies from conda to "pip" section

## How is this patch tested?

N/A

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/fastai.py [hunk 1]
```diff
@@ -77,11 +77,9 @@ def get_default_conda_env(include_cloudpickle=False):
     pip_deps = None
     if include_cloudpickle:
         import cloudpickle
-
         pip_deps = ["cloudpickle=={}".format(cloudpickle.__version__)]
     return _mlflow_conda_env(
-        additional_conda_deps=["fastai={}".format(fastai.__version__)],
-        additional_pip_deps=pip_deps,
+        additional_pip_deps=["fastai={}".format(fastai.__version__)] + pip_deps,
```
Comment (harupy) [inline mlflow/fastai.py:82]: I think this line fails when `pip_deps` is None. Can we fix the code here to the following?

```python
    pip_deps = ["fastai={}".format(fastai.__version__)]
    if include_cloudpickle:
        import cloudpickle
        pip_deps.append("cloudpickle=={}".format(cloudpickle.__version__))

    return _mlflow_conda_env(
        additional_pip_deps=pip_deps,
        additional_conda_channels=None,
    )
```

File: mlflow/fastai.py [hunk 2]
```diff
@@ -74,16 +74,12 @@ def get_default_conda_env(include_cloudpickle=False):
...
         import cloudpickle
 
-        pip_deps = ["cloudpickle=={}".format(cloudpickle.__version__)]
-    return _mlflow_conda_env(
-        additional_conda_deps=["fastai={}".format(fastai.__version__)],
-        additional_pip_deps=pip_deps,
-        additional_conda_channels=None,
-    )
+        pip_deps.append("cloudpickle=={}".format(cloudpickle.__version__))
+    return _mlflow_conda_env(additional_pip_deps=pip_deps, additional_conda_channels=None,)
```
Comment (harupy) [inline mlflow/fastai.py:82]:
```suggestion
    return _mlflow_conda_env(additional_pip_deps=pip_deps)
```

@WeichenXu123 Could you clean up useless arguments like this one?

File: mlflow/utils/environment.py [hunk 3]
```diff
@@ -5,48 +5,36 @@
...
 """
 
 
 def _mlflow_conda_env(
-    path=None,
-    additional_conda_deps=None,
-    additional_pip_deps=None,
-    additional_conda_channels=None,
-    install_mlflow=True,
+    path=None, additional_pip_deps=None, install_mlflow=True,
```
Comment (harupy) [inline mlflow/utils/environment.py:13]: Should we avoid removing `additional_conda_deps` and `additional_conda_deps` because it's a breaking change?

Comment (harupy) [reply mlflow/utils/environment.py:13]: We could depreate them in MLflow 1.1.8.0 and removed them in MLflow 1.19.0:
> `additional_conda_deps` and `additional_conda_deps` will be removed in the future.

Comment (WeichenXu123) [reply mlflow/utils/environment.py:13]: i think it's not an public API

File: mlflow/utils/environment.py [hunk 4]
```diff
@@ -5,48 +5,36 @@
 _conda_header = """\
 name: mlflow-env
 channels:
-  - defaults
```
[the remark this answers is not in the document]
Comment (WeichenXu123) [reply mlflow/utils/environment.py:8]: https://databricks.atlassian.net/browse/ML-15265
proposed here

[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/utils/environment.py:8]: Ah I missed that

File: mlflow/utils/environment.py (hunk 3 above)
Comment (harupy) [reply mlflow/utils/environment.py]: Offline discussion: we don't have to remove these arguments for now.

File: mlflow/utils/environment.py [hunk 5]
```diff
@@ -5,48 +5,40 @@
...
 
 
-def _mlflow_conda_env(
-    path=None,
-    additional_conda_deps=None,
-    additional_pip_deps=None,
-    additional_conda_channels=None,
-    install_mlflow=True,
-):
+def _mlflow_conda_env(path=None, additional_pip_deps=None, install_mlflow=True, pip_version=None):
```
Comment (harupy) [inline mlflow/utils/environment.py:12]: As we discussed today, could you revert this change?

File: mlflow/pytorch/__init__.py [hunk 6]
```diff
@@ -69,26 +69,25 @@ def get_default_conda_env():
...
+                                             'cloudpickle==1.6.0']}]}
     """
     import torch
     import torchvision
 
     return _mlflow_conda_env(
-        additional_conda_deps=[
+        additional_pip_deps=[
             "pytorch={}".format(torch.__version__),
             "torchvision={}".format(torchvision.__version__),
```
Comment (harupy) [inline mlflow/pytorch/__init__.py:85]:
```suggestion
            "pytorch=={}".format(torch.__version__),
            "torchvision=={}".format(torchvision.__version__),
```

Should we use `==` instead of `=`?

**Review by harupy — COMMENTED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/4393/files#r640267418 is addressed

File: mlflow/pytorch/__init__.py [hunk 7]
```diff
@@ -69,26 +69,25 @@ def get_default_conda_env():
...
-                   'channels': ['defaults', 'conda-forge', 'pytorch'],
-                   'dependencies': ['python=3.7.5', 'pytorch=1.5.1',
-                                    'torchvision=0.6.1',
-                                    'pip', {'pip': ['mlflow', 'cloudpickle==1.6.0']}]}
+                   'channels': ['conda-forge'],
+                   'dependencies': ['python=3.7.5',
+                                    {'pip': ['pytorch=1.5.1',
+                                             'torchvision=0.6.1',
+                                             'mlflow',
+                                             'cloudpickle==1.6.0']}]}
```
Comment (harupy) [inline mlflow/pytorch/__init__.py:77]:
```suggestion
                                    {'pip': ['pytorch==1.5.1',
                                             'torchvision==0.6.1',
                                             'mlflow',
                                             'cloudpickle==1.6.0']}]}
```

Relate to https://github.com/mlflow/mlflow/pull/4393/files#r640267418

File: mlflow/statsmodels.py [hunk 8]
```diff
@@ -55,11 +55,7 @@ def get_default_conda_env():
     """
     import statsmodels
 
-    return _mlflow_conda_env(
-        additional_conda_deps=["statsmodels={}".format(statsmodels.__version__)],
-        additional_pip_deps=None,
-        additional_conda_channels=None,
-    )
+    return _mlflow_conda_env(additional_pip_deps=["statsmodels={}".format(statsmodels.__version__)])
```
Comment (harupy) [inline mlflow/statsmodels.py:58]:
```suggestion
    return _mlflow_conda_env(additional_pip_deps=["statsmodels=={}".format(statsmodels.__version__)])
```

File: mlflow/tensorflow.py [hunk 9]
```diff
@@ -74,11 +74,7 @@ def get_default_conda_env():
     """
     import tensorflow
 
-    return _mlflow_conda_env(
-        additional_conda_deps=["tensorflow={}".format(tensorflow.__version__)],
-        additional_pip_deps=None,
-        additional_conda_channels=None,
-    )
+    return _mlflow_conda_env(additional_pip_deps=["tensorflow={}".format(tensorflow.__version__)])
```
Comment (harupy) [inline mlflow/tensorflow.py:77]:
```suggestion
    return _mlflow_conda_env(additional_pip_deps=["tensorflow=={}".format(tensorflow.__version__)])
```

**Review by harupy — APPROVED** [review]
LGTM
